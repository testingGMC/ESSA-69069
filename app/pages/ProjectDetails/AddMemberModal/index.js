import { Form } from 'antd'
import AddMemberModal from './AddMemberModal'
export default Form.create({ name: 'add_member' })(AddMemberModal)
