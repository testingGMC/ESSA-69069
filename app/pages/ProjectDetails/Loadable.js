/**
 *
 * Asynchronously loads the component for ProjectDetails
 *
 */

import loadable from 'utils/loadable'

export default loadable(() => import('./index'))
