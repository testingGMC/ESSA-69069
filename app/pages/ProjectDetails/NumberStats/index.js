import NumberStats from './NumberStats'

export default NumberStats
