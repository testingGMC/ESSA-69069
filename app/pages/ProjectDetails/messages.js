/*
 * ProjectDetails Messages
 *
 * This contains all the text for the ProjectDetails container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.ProjectDetails'

export default defineMessages({
  projectBoard: {
    id: `${scope}.projectBoard`,
    defaultMessage: 'Project Board',
  },
  startDate: {
    id: `${scope}.startDate`,
    defaultMessage: 'start Date',
  },
  dueDate: {
    id: `${scope}.dueDate`,
    defaultMessage: 'Due Date',
  },
  startDateEndDate: {
    id: `${scope}.startDateEndDate`,
    defaultMessage: ' Start Date {startDate} | Due Date {dueDate}',
  },
  description: {
    id: `${scope}.description`,
    defaultMessage: ' { description }',
  },
  projectTitle: {
    id: `${scope}.projectTitle`,
    defaultMessage: 'About the project',
  },
  stats: {
    id: `${scope}.stats`,
    defaultMessage: 'Stats',
  },
  efficiency: {
    id: `${scope}.efficiency`,
    defaultMessage: 'Efficiency',
  },
  progress: {
    id: `${scope}.progress`,
    defaultMessage: 'Progress',
  },
  estimatedProgress: {
    id: `${scope}.estimatedProgress`,
    defaultMessage: 'Estimated Progress',
  },
  teamMember: {
    id: `${scope}.teamMember`,
    defaultMessage: 'Team members',
  },
  addNewMember: {
    id: `${scope}.addNewMember`,
    defaultMessage: 'Add new Member +',
  },
  delete: {
    id: `${scope}.delete`,
    defaultMessage: 'Delete',
  },

  confirmMessage: {
    id: `${scope}.confirmMessage`,
    defaultMessage: 'Are you sure to delete this member?',
  },
  yes: {
    id: `${scope}.yes`,
    defaultMessage: 'Yes',
  },
  no: {
    id: `${scope}.no`,
    defaultMessage: 'No',
  },
  numbers: {
    id: `${scope}.numbers`,
    defaultMessage: 'Numbers',
  },
  numberOfTasks: {
    id: `${scope}.numberOfTasks`,
    defaultMessage: 'Number Of Tasks',
  },
  completedTasks: {
    id: `${scope}.completedTasks`,
    defaultMessage: 'Completed Tasks',
  },
  nbOfRemainingDays: {
    id: `${scope}.nbOfRemainingDays`,
    defaultMessage: 'Nb. of remaining days',
  },
  helmetDescription: {
    id: `${scope}.helmetDescription`,
    defaultMessage: 'Description of Project Details',
  },
  helmetTitle: {
    id: `${scope}.helmetTitle`,
    defaultMessage: 'Project Details',
  },
})
