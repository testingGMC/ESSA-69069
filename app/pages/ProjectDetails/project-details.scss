.project-details {
    padding: 27px 23px 47px 35px;

    .row-title {
        margin-bottom: 27px;

        .ant-page-header {
            display: inline-flex;
            align-items: center;
            padding: 16px 4px;

            .ant-page-header-title-view-title {
                font-size: 28px;
            }
        }
    }

    .ant-row {
        width: 100%;
    }

    .project-title {
        font-size: 28px;
        font-weight: bold;

        padding-right: 24px;
    }

    .project-icon {
        font-size: 20px;
        color: #cecece;
        display: inline-block;
        position: relative;
        color: #cecece;
        cursor: pointer;
    }

    .project-board  {
        a{
        text-align: right;
        color: #979797;
        font-size: 18px;
        font-weight: 600;
        display: flex;
        justify-content: flex-end;
        align-items: center;
        }
        .link {
            margin-right: 20px;
        }
    }

    .row-title {
        align-items: center;
    }

    .project-content,
    .stats-content {
        height: 200px;
    }

    .period {
        color: #aeaeae;
        font-size: 13px;
        font-weight: 400;
        margin-bottom: 5px;
    }

    .custom-row {
        margin-top: 24px;
    }

    .team-members {
        .team-members-extra {
            .add-member {
                font-size: 13px;
                font-weight: 600;
                margin-right: 40px;
                color: #979797;
                cursor: pointer;
                margin-left: 40px;
            }

            button {

                width: 26px;
                height: 26px;
                background-color: #ececec;

                &.ant-btn-circle {
                    font-size: 13px !important;
                }
            }
        }

        .ant-card-body {
            overflow: auto;
            height: 300px;

            &::-webkit-scrollbar {
                width: 3px;
            }

            &::-webkit-scrollbar-track {
                background-color: white;
            }

            &::-webkit-scrollbar-thumb {
                background-color: #2a7bdb;
                outline: 1px solid slategrey;
                border-radius: 50px 50px;
            }
        }
    }

    .with-margin-bottom {
        margin-bottom: 24px;
    }

    .no-members {
        display: flex;
        align-items: center;
        justify-content: center;
        height: 250px;
    }
}