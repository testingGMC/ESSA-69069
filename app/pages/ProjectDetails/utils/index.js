export const getFilteredMembers = (members, searchValue) => {
  const value = searchValue.toLowerCase()
  return members.filter(m =>
    !m.fullName ? true : m.fullName.toLowerCase().includes(value),
  )
}
