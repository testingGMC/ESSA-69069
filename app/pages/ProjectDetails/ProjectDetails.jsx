/* eslint-disable react/no-array-index-key */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable global-require */
/**
 *
 * ProjectDetails
 *
 */
import QueueAnim from 'rc-queue-anim'
import React, { useState, useEffect } from 'react'
import { Helmet } from 'react-helmet'
import PropTypes from 'prop-types'
import moment from 'moment'
import { isEmpty } from 'lodash'
import { Link } from 'react-router-dom'
import Texty from 'rc-texty'
import {
  FormattedMessage,
  FormattedDate,
  injectIntl,
  intlShape,
} from 'react-intl'
import Card from 'shared/components/Card'
import UserCard from 'shared/components/UserCard'
import routes from 'shared/routes'
import {
  Menu,
  Row,
  Col,
  Skeleton,
  Popconfirm,
  Empty,
  Input,
  Icon,
  PageHeader,
} from 'antd'
import CircularProgress from 'shared/components/CircularProgress'

import { unflattenArray } from 'shared/utils/array.helper'
import ProjectModal from 'pages/Dashboard/ProjectModal'
import { isContentArchitect, getUserId } from 'shared/utils/access-token'
import messages from './messages'

import NumberStats from './NumberStats'
import AddMemberModal from './AddMemberModal'
import { getFilteredMembers } from './utils'
import './project-details.scss'
import { select } from 'redux-saga/effects'

const NUMBER_IMAGES = [
  'detail-img1.png',
  'detail-img2.png',
  'detail-img3.png',
  'detail-img4.png',
]
const PostDate = injectIntl(({ date, intl }) => (
  <span title={intl.formatDate(date)}>
    <FormattedDate value={date} year="numeric" month="long" day="2-digit" />
  </span>
))

const ProjectDetails = ({
  deleteMember,
  match: {
    params: { projectId },
  },
  fetchProjectById,
  fetchProjectStatsById,
  addMember,
  updateProject,
  selectedProject,
  roles,
  intl,
  setIsAddMemberModalVisible,
  setIsUpdateModalVisible,
}) => {
  const { isAddMemberModalVisible, isUpdateModalVisible } = selectedProject
  const [formRef, setFormRef] = useState(null)
  const [searchValue, setSearchValue] = useState('')
  const [members, setMembers] = useState(
    selectedProject.data && selectedProject.data.members
      ? selectedProject.data.members
      : [],
  )

  useEffect(() => {
    fetchProjectById(projectId)
  }, [])

  useEffect(() => {
    setMembers(
      selectedProject.data && selectedProject.data.members
        ? getFilteredMembers(selectedProject.data.members, searchValue)
        : [],
    )
  }, [selectedProject, searchValue])

  const handleAddMember = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      const member = {
        memberId: values.memberId,
        projectId,
      }
      addMember(member)
    })
  }
  const handleUpdateProject = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      const project = {
        name: values.name,
        description: values.description,
        startDate: selectedProject.data.startDate,
        dueDate: values.dueDate,
        members: selectedProject.data.members,
      }
      updateProject(projectId, project)
    })
  }

  const handleAddMemberModalClose = () => {
    setIsAddMemberModalVisible(false)
  }
  const handleUpdateProjectModalClose = () => {
    setIsUpdateModalVisible(false)
  }

  const myMenu = memberId => (
    <Menu>
      <Menu.Item>
        <Popconfirm
          placement="topLeft"
          title={<FormattedMessage {...messages.confirmMessage} />}
          onConfirm={() => deleteMember(memberId)}
          okText={<FormattedMessage {...messages.yes} />}
          cancelText={<FormattedMessage {...messages.no} />}
        >
          <a href="#">{<FormattedMessage {...messages.delete} />}</a>
        </Popconfirm>
      </Menu.Item>
    </Menu>
  )
  const renderMembers = () => {
    const result = []
    if (isEmpty(members)) {
      return (
        <div className="no-members">
          <Empty />
        </div>
      )
    }
    unflattenArray(members, 2).forEach((rowMember, index) => {
      result.push(
        <Row key={index} gutter={24}>
          <QueueAnim type={['right', 'left']} leaveReverse>
            {rowMember.map((member, idx) => (
              <Col
                xs={24}
                sm={24}
                md={12}
                lg={12}
                xl={12}
                key={member.id}
                className="with-margin-bottom"
              >
                <div className="member-item">
                  <UserCard
                    key={idx}
                    active={member.active}
                    title={`${member.fullName}`}
                    subTitle={member.role}
                    menu={myMenu(member.id)}
                    canDelete={
                      member.memberId.toString() !==
                      selectedProject.data.createdBy.toString()
                    }
                    roles={roles}
                  />
                </div>
              </Col>
            ))}
          </QueueAnim>
        </Row>,
      )
    })
    return result
  }
  const saveFormRef = formRefParameter => {
    setFormRef(formRefParameter)
  }
  const helmetMessages = {
    helmetTitle: intl.formatMessage({
      ...messages.helmetTitle,
    }),
    helmetDescription: intl.formatMessage({
      ...messages.helmetDescription,
    }),
  }
  return (
    <div className="project-details">
      <Helmet>
        <title>{helmetMessages.helmetTitle}</title>
        <meta name="description" content={helmetMessages.helmetDescription} />
      </Helmet>
      <div className="project-title">
        {selectedProject.data && (
          <>
            <Row gutter={24} type="flex" className="row-title">
              <Col xs={24} sm={12} md={12} lg={12} xl={12}>
                <Skeleton
                  loading={selectedProject.loading}
                  active
                  paragraph={false}
                >
                  <PageHeader
                    onBack={() => {
                      window.history.back()
                    }}
                    title={<Texty>{selectedProject.data.name}</Texty>}
                  />

                  {isContentArchitect(roles) && (
                    <span className="project-icon">
                      <Icon
                        type="edit"
                        theme="filled"
                        onClick={() => setIsUpdateModalVisible(true)}
                      />
                    </span>
                  )}
                </Skeleton>
              </Col>
              <Col xs={24} sm={12} md={12} lg={12} xl={12}>
                <div className="project-board">
                  {isContentArchitect(roles) &&
                    (selectedProject.data &&
                      selectedProject.data.members &&
                      selectedProject.data.members.find(
                        member =>
                          member.memberId.toString() === getUserId().toString(),
                      )) && (
                    <Link to={`${routes.SPRINT_BOARD.linkPath(projectId)}`}>
                      <span className="link">
                        <FormattedMessage {...messages.projectBoard} />
                      </span>
                      <Icon type="arrow-right" />
                    </Link>
                  )}
                </div>
              </Col>
            </Row>
          </>
        )}
      </div>

      <Row gutter={24}>
        <Col
          xs={24}
          sm={24}
          md={24}
          lg={12}
          xl={12}
          className="with-margin-bottom"
        >
          <Card title={<FormattedMessage {...messages.projectTitle} />}>
            <Skeleton loading={selectedProject.loading} active>
              {selectedProject.data && (
                <div className="project-content">
                  <div className="period">
                    <FormattedMessage
                      {...messages.startDateEndDate}
                      values={{
                        startDate: (
                          <PostDate date={selectedProject.data.startDate} />
                        ),
                        dueDate: (
                          <PostDate date={selectedProject.data.dueDate} />
                        ),
                      }}
                    />
                  </div>
                  <FormattedMessage
                    {...messages.description}
                    values={{
                      description: (
                        <Texty>{selectedProject.data.description}</Texty>
                      ),
                    }}
                  />
                </div>
              )}
            </Skeleton>
          </Card>
        </Col>
        <Col
          xs={24}
          sm={24}
          md={24}
          lg={12}
          xl={12}
          className="with-margin-bottom"
        >
          <Card
            title={<FormattedMessage {...messages.stats} />}
            extra={
              <Icon
                type="reload"
                onClick={() => fetchProjectStatsById(projectId)}
              />
            }
          >
            <Skeleton
              loading={selectedProject.loading || selectedProject.loadingStats}
              active
            >
              {selectedProject.data && selectedProject.data.stats && (
                <div className="stats-content">
                  <Row gutter={24}>
                    <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                      <CircularProgress
                        percentage={
                          selectedProject.data.stats.estimatedProgress * 100
                        }
                        text={`${Number(
                          selectedProject.data.stats.estimatedProgress * 100,
                        ).toFixed(0)}%`}
                        title={
                          <FormattedMessage {...messages.estimatedProgress} />
                        }
                        pathColor="#f5a13e"
                      />
                    </Col>
                    <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                      <CircularProgress
                        percentage={selectedProject.data.stats.progress * 100}
                        text={`${Number(
                          selectedProject.data.stats.progress * 100,
                        ).toFixed(0)}%`}
                        title={<FormattedMessage {...messages.progress} />}
                        pathColor="#7ed327"
                      />
                    </Col>
                  </Row>
                </div>
              )}
            </Skeleton>
          </Card>
        </Col>
      </Row>
      <Row gutter={24}>
        <Col
          xs={24}
          sm={24}
          md={24}
          lg={12}
          xl={12}
          className="team-members with-margin-bottom"
        >
          <Card
            title={<FormattedMessage {...messages.teamMember} />}
            extra={
              <div className="team-members-extra">
                {isContentArchitect(roles) &&
                  (selectedProject.data &&
                    selectedProject.data.members &&
                    selectedProject.data.members.find(
                      member =>
                        member.memberId.toString() === getUserId().toString(),
                    )) && (
                    <span
                      className="add-member"
                      onClick={() => setIsAddMemberModalVisible(true)}
                    >
                      {<FormattedMessage {...messages.addNewMember} />}
                    </span>
                  )}
                <Input.Search
                  placeholder=""
                  onChange={e => setSearchValue(e.target.value)}
                  style={{ width: 150 }}
                />
              </div>
            }
            className="team-members-content"
          >
            <Skeleton loading={selectedProject.loading} avatar active>
              {selectedProject.data && <div>{renderMembers()}</div>}
            </Skeleton>
          </Card>
        </Col>
        <Col xs={24} sm={24} md={24} lg={12} xl={12}>
          <Card title={<FormattedMessage {...messages.numbers} />}>
            <Skeleton
              loading={selectedProject.loading || selectedProject.loadingStats}
              active
            >
              {selectedProject.data && selectedProject.data.stats && (
                <>
                  <QueueAnim type={['right', 'left']} leaveReverse>
                    <Row gutter={24} className="numbers-items" key={0}>
                      <Col
                        xs={24}
                        sm={12}
                        md={12}
                        lg={12}
                        xl={12}
                        className="numbers-item detail with-margin-bottom"
                      >
                        <NumberStats
                          key={1}
                          image={require(`../../assets/icons/${
                            NUMBER_IMAGES[0]
                          }`)}
                          number={selectedProject.data.stats.numberOfTasks.toString()}
                          text={
                            <FormattedMessage {...messages.numberOfTasks} />
                          }
                        />
                      </Col>
                      <Col
                        xs={24}
                        sm={12}
                        md={12}
                        lg={12}
                        xl={12}
                        className="numbers-item detail with-margin-bottom"
                      >
                        <NumberStats
                          key={2}
                          image={require(`../../assets/icons/${
                            NUMBER_IMAGES[1]
                          }`)}
                          number={selectedProject.data.stats.numberOfCompletedTasks.toString()}
                          text={
                            <FormattedMessage {...messages.completedTasks} />
                          }
                        />
                      </Col>
                    </Row>
                    <Row gutter={24} className="numbers-items" key={1}>
                      <Col
                        xs={24}
                        sm={12}
                        md={12}
                        lg={12}
                        xl={12}
                        className="numbers-item detail with-margin-bottom"
                      >
                        <NumberStats
                          key={3}
                          image={require(`../../assets/icons/${
                            NUMBER_IMAGES[2]
                          }`)}
                          number={moment(selectedProject.data.dueDate)
                            .diff(selectedProject.data.startDate, 'days')
                            .toString()}
                          text={
                            <FormattedMessage {...messages.nbOfRemainingDays} />
                          }
                        />
                      </Col>
                      <Col
                        xs={24}
                        sm={12}
                        md={12}
                        lg={12}
                        xl={12}
                        className="numbers-item detail with-margin-bottom"
                      >
                        <NumberStats
                          key={4}
                          image={require(`../../assets/icons/${
                            NUMBER_IMAGES[3]
                          }`)}
                          number={Number(
                            selectedProject.data.stats.efficiency,
                          ).toFixed(2)}
                          text={<FormattedMessage {...messages.efficiency} />}
                        />
                      </Col>
                    </Row>
                  </QueueAnim>
                </>
              )}
            </Skeleton>
          </Card>
        </Col>
      </Row>
      {isAddMemberModalVisible && (
        <AddMemberModal
          wrappedComponentRef={saveFormRef}
          isVisible={isAddMemberModalVisible}
          handleCancel={handleAddMemberModalClose}
          handleOk={handleAddMember}
          membersIds={members.map(member => member.memberId)}
          fetch
        />
      )}
      {isUpdateModalVisible && (
        <ProjectModal
          wrappedComponentRef={saveFormRef}
          project={selectedProject.data}
          isVisible={isUpdateModalVisible}
          handleCancel={handleUpdateProjectModalClose}
          handleOk={handleUpdateProject}
        />
      )}
    </div>
  )
}
ProjectDetails.propTypes = {
  selectedProject: PropTypes.object.isRequired,
  deleteMember: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  fetchProjectById: PropTypes.func.isRequired,
  fetchProjectStatsById: PropTypes.func.isRequired,
  addMember: PropTypes.func.isRequired,
  updateProject: PropTypes.func.isRequired,
  roles: PropTypes.string.isRequired,
  intl: intlShape.isRequired,
  setIsAddMemberModalVisible: PropTypes.func.isRequired,
  setIsUpdateModalVisible: PropTypes.func.isRequired,
}

export default injectIntl(ProjectDetails)
