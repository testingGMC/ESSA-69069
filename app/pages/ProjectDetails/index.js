/**
 *
 * ProjectDetails
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { useInjectReducer } from 'utils/injectReducer'
import makeSelectProjectDetails from 'shared/redux/selectedproject/selectors'
import reducer from 'shared/redux/selectedproject/reducer'

import * as actions from 'shared/redux/selectedproject/actions'

import saga from 'shared/redux/selectedproject/saga'
import { DAEMON } from 'utils/constants'
import ProjectDetails from './ProjectDetails'

const ProjectDetailsIndex = props => {
  useInjectReducer({ key: 'selectedProject', reducer })
  return <ProjectDetails {...props} />
}
const withSaga = injectSaga({ key: 'selectedProject', saga, mode: DAEMON })

const mapStateToProps = createStructuredSelector({
  selectedProject: makeSelectProjectDetails(),
})

const mapDispatchToProps = {
  ...actions,
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  withSaga,
  memo,
)(ProjectDetailsIndex)
