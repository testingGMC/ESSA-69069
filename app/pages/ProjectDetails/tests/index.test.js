import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'

import ProjectDetailsIndex from '../index'
import ProjectDetails from '../ProjectDetails'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'

const initialProps = {
  deleteMember: jest.fn(),
  fetchProjectById: jest.fn(),
  fetchProjectStatsById: jest.fn(),
  addMember: jest.fn(),
  updateProject: jest.fn(),
  selectedProject: { loading: false },
  match: {
    params: { projectId: '1' },
  },
}

describe('<ProjectDetailsIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ProjectDetailsIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ProjectDetailsIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })
})
describe('<ProjectDetails />', () => {
  let store

  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ProjectDetails {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()
  })

  it('should fetch the selectedProject on mount if the projectId exists', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ProjectDetails {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(initialProps.fetchProjectById).toHaveBeenCalled()
  })
})
