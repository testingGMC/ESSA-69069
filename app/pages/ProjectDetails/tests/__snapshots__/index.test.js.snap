// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectDetails /> should render and match the snapshot 1`] = `
<div
  class="project-details"
>
  <div
    class="project-title"
  />
  <div
    class="ant-row"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col with-margin-bottom ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-12 ant-col-xl-12"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="card"
      >
        <div
          class="ant-card ant-card-small"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <span>
                  About the project
                </span>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="content "
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col with-margin-bottom ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-12 ant-col-xl-12"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="card"
      >
        <div
          class="ant-card ant-card-small"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <span>
                  Stats
                </span>
              </div>
              <div
                class="ant-card-extra"
              >
                <i
                  aria-label="icon: reload"
                  class="anticon anticon-reload"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="reload"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 0 0-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 0 1 655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 0 1 279 755.2a342.16 342.16 0 0 1-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 0 1 109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 0 0 3 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
                    />
                  </svg>
                </i>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="content "
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col team-members with-margin-bottom ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-12 ant-col-xl-12"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="card"
      >
        <div
          class="ant-card ant-card-small"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <span>
                  Team members
                </span>
              </div>
              <div
                class="ant-card-extra"
              >
                <div
                  class="team-members-extra"
                >
                  <span
                    class="ant-input-search ant-input-affix-wrapper"
                    style="width: 150px;"
                  >
                    <input
                      class="ant-input"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <i
                        aria-label="icon: search"
                        class="anticon anticon-search ant-input-search-icon"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="search"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                          />
                        </svg>
                      </i>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="content team-members-content"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-12 ant-col-xl-12"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="card"
      >
        <div
          class="ant-card ant-card-small"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <span>
                  Numbers
                </span>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="content "
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
